system/jlib/jsocket.cpp

```diff
@@ -5213,15 +5213,47 @@ class CSocketEpollThread: public CSocketBaseThread
         }
     }
 
+    // JCSMORE: this is very similar to the version in CSocketSelectThread. Could be commoned if items used common structure.
+    void updateItems()
+    {
+        // must be in CriticalBlock block(sect);
+        unsigned n = items.ordinality();
+        while (n)
+        {
+            n--;
+            SelectItem *si = items.element(n);
+            if (si->del)
+            {
+                // Release/dtors should not throw but leaving try/catch here until all paths checked
+                try
+                {
+#ifdef SOCKTRACE
+                    PROGLOG("CSocketSelectThread::updateItems release %d",si.handle);
+#endif
+                    si->nfy->Release();
+                    si->sock->Release();
+                    delete si;
+                }
+                catch (IException *e)
+                {
+                    EXCLOG(e,"CSocketSelectThread::updateItems");
+                    e->Release();
+                }
+                items.remove(n);
```

Comment:
Is it worth keeping the logic from delSelItemPos - swap this with the last, and then remove the last item.  May even be a useful method in the array class - since it saves moving all the following elements in memory.

Reply:
As it is now, it's walking the array backwards, and isn't having to shift anything